pkg/datapath/linux/node.go

```diff
@@ -630,6 +644,73 @@ func (n *linuxNodeHandler) insertNeighbor(newNode *nodeTypes.Node, ifaceName str
 	}
 }
 
+// Must be called with linuxNodeHandler.mutex held.
+func (n *linuxNodeHandler) refreshNeighbor(nodeToRefresh *nodeTypes.Node, ifaceName string) {
```

Comment:
There is a bit of duplication with `insertNeighbor()` method. Perhaps it makes sense to merge both into one?